pkg/kademlia/workers.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package kademlia
+
+import (
+	"container/heap"
+	"context"
+	"log"
+	"math/big"
+	"sync"
+	"time"
+
+	"storj.io/storj/pkg/dht"
+
+	"github.com/zeebo/errs"
+	"storj.io/storj/pkg/node"
+	proto "storj.io/storj/protos/overlay"
+)
+
+var (
+	// WorkerError is the class of errors for the worker struct
+	WorkerError = errs.Class("worker error")
+	// default timeout is the minimum timeout for worker cancellation
+	// 250ms was the minimum value allowing current workers to finish work
+	// before returning
+	defaultTimeout = 250 * time.Millisecond
+)
+
+type worker struct {
+	pq             PriorityQueue
+	mu             *sync.Mutex
+	maxResponse    time.Duration
+	cancel         context.CancelFunc
+	nodeClient     node.Client
+	find           dht.NodeID
+	workInProgress int
+	k              int
+}
+
+func newWorker(ctx context.Context, rt *RoutingTable, nodes []*proto.Node, nc node.Client, target dht.NodeID, k int) *worker {
+	t := new(big.Int).SetBytes(target.Bytes())
+
+	pq := func(nodes []*proto.Node) PriorityQueue {
```

Comment:
why the anonymous immediately called function? you could do a block with just normal {} braces

Reply:
It seemed cleaner to me, your prefer it to not have an anonymous function ?

Reply:
it's fine, though you can just do a subscope (e.g. https://medium.com/golangspec/scopes-in-go-a6042bb4298c)